dtfs-central-api/src/types/utilisation-reports.d.ts

```diff
@@ -48,3 +49,9 @@ export type UpdateUtilisationReportStatusInstructions = {
   status: UtilisationReportReconciliationStatus;
   filter: ReportFilter;
 };
+
+type UtilisationReportHeader = ValuesOf<typeof UTILISATION_REPORT_HEADERS>;
+
+export type ReportData = {
+  [P in UtilisationReportHeader]: P extends `${string}currency` ? Currency : string;
```

Comment:
You could change this to match on `${string}currency${string}` which will then cover `currency` being anywhere in the string (including both the start and end, as `${string}` could be the empty string).

Reply:
I didn't do that specifically because there could be a key like "X currency exchange rate", which would not be a `Currency`. I thought it was more logical that any string ending with "currency" would probably itself be a `Currency`

Reply:
Ah, yeah fair point.